cns/nodesubnet/ip_fetcher_test.go

```diff
@@ -12,55 +13,117 @@ import (
 
 // Mock client that simply tracks if refresh has been called
 type TestClient struct {
-	fetchCalled bool
+	refreshCount int32
 }
 
 // Mock refresh
 func (c *TestClient) GetInterfaceIPInfo(_ context.Context) (nmagent.Interfaces, error) {
-	c.fetchCalled = true
+	atomic.AddInt32(&c.refreshCount, 1)
```

Comment:
Atomics are pretty rare unless there's a specific performance need. I forget the exact reasoning, but there's some sharp edges on them if I recall. Usually a mutex is fine.

Reply:
Switched to mutex.